### PR TITLE
fix duplicated error messages

### LINKED ISSUE #79: Showing duplicated errors
**Current behavior**
The same error is being shown multiple times

**Expected behavior**
If a specific error happens multiple times, show it only once

**Steps to reproduce it**
1. Select "primitives" > "booleanPrimitives" as the source
1. Select "addressList" > "city" as the target (error is shown)
1. Select "addressList" > "city" as the target again (second error with the same message is shown)
1. Select "addressList" > "city" as the target again (third error with the same message is shown)

### LINKED ISSUE #679: [Datamapper]Error message for conversion are stacking up
If I click several times between the same elements in the lists (and so just wanting to highlight the same mapping and if it is invalid, the error message is stacking up:

image

### PR DESCRIPTION
**Pull request by abkieling**
fix #79 
fix [https://github.com/syndesisio/syndesis-ui/issues/679](https://github.com/syndesisio/syndesis-ui/issues/679)

### CODE REVIEW

File: src/app/lib/syndesis-data-mapper/services/error.handler.service.ts [hunk 1]
```diff
@@ -27,36 +27,30 @@ export class ErrorHandlerService {
     public info(message: string, error: any) { this.addError(message, ErrorLevel.INFO, error); }
     public warn(message: string, error: any) { this.addError(message, ErrorLevel.WARN, error); }
     public error(message: string, error: any) { this.addError(message, ErrorLevel.ERROR, error); }
-    public validationError(message: string, error: any) { this.addError(message, ErrorLevel.VALIDATION_ERROR, error); }
+    public validationError(message: string, error: any) { this.addValidationError(message, ErrorLevel.VALIDATION_ERROR, error); }
```
Comment (rhuss) [inline src/app/lib/syndesis-data-mapper/services/error.handler.service.ts]: Not really in the code, but why using both, a method name 'addValidationError' and an `ErrorLevel.VALIDATION_ERROR`. Feels redundant to me (e.g. when the method is only about validation errors, then you don't need the second argument as you can set the level within the method itself)

**Review by rhuss — COMMENTED** [review]
LGTM (with one minor comment)

File: src/app/lib/syndesis-data-mapper/services/error.handler.service.ts (hunk 1 above)
Comment (abkieling) [reply src/app/lib/syndesis-data-mapper/services/error.handler.service.ts]: Fair point. I'll change that.

**Comment by abkieling** [comment]
@rhuss made the change you requested

**Review by gashcrumb — APPROVED** [review]

**Pull request merged**